web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/CdsProfileFetcher.java

```diff
@@ -133,7 +133,18 @@ private ProfileFetcherResult internalFetchAppProfile(String instrumentationKey,
         // if no task currently exists for this ikey, then let's create one.
         if (currentTask == null) {
             currentTask = createFetchTask(instrumentationKey, configuration);
-            this.tasks.putIfAbsent(instrumentationKey, currentTask);
+            Future<HttpResponse> mapValue = this.tasks.putIfAbsent(instrumentationKey, currentTask);
+            // it's possible for another thread to create a fetch task
+            if (mapValue != null) {
+                if (currentTask.isDone() && !mapValue.isDone()) {
+                    // if the new task completed before the old task somehow, use that one and cancel the old
+                    mapValue.cancel(true);
```

Comment:
makes sense. we can just ignore this alert

Reply:
Instead of trying to determine which one is done and canceling the other, I'm just going to let the new one "evaporate" (since it wouldn't be added to the map) and set `currentTask = mapValue` since `mapValue` has already been running. Then everything else will work normally.